### PR TITLE
Remove `modules/transport_netty_3` in favor of `netty_4`

### PR DESCRIPTION
**Pull request by s1monw**
We kept `netty_3` as a fallback in the 5.x series but now that master
is 6.0 we don't need this or in other words all issues coming up with
netty 4 will be blockers for 6.0.

### CODE REVIEW

**Review by ywelsch — APPROVED** [review]
bye bye netty3 💃

**Comment by nik9000** [comment]
Is it worth running integ-test-zip tests with a two node cluster so we have something that does it? The backwards tests also run a two node cluster but that seems like a different test.

File: modules/transport-netty4/build.gradle [hunk 1]
```diff
@@ -42,13 +42,6 @@ dependencies {
     compile "io.netty:netty-transport:4.1.6.Final"
 }
 
-integTest {
-    includePackaged = true
-    cluster {
-        numNodes = 2
```
Comment (jasontedor) [inline modules/transport-netty4/build.gradle]: I think that this should stay.

**Review by jasontedor — CHANGES_REQUESTED** [review]
LGTM, except for one thing.

File: modules/transport-netty4/build.gradle (hunk 1 above)
Comment (jasontedor) [inline modules/transport-netty4/build.gradle]: What I mean is, I think we should run the regular REST tests with two nodes.

Comment (s1monw) [inline modules/transport-netty4/build.gradle]: there are no regular rest tests here? there is only the one that checks that the plugin is loaded?

Comment (jasontedor) [inline modules/transport-netty4/build.gradle]: I understand that; we use to run the regular REST tests here with two nodes. I'm saying we should still run the regular REST tests with two nodes somewhere, probably as part of the regular distribution integration test. This seemed the best place to leave that thought. :smile:

Comment (s1monw) [inline modules/transport-netty4/build.gradle]: maybe `qa/smoke-test-multinode`?

Comment (nik9000) [inline modules/transport-netty4/build.gradle]: Ah! Fine then. I think we should remove that one in favor of running with two nodes by default but that can be a thing to another PR.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by tlrx** [comment]
Shouldn't we also adapt the bwc tests so that it does not randomize the transport type when executing bwc tests against a future 6.x minor version (like testing 6.1.0 against 6.0.1)?

**Comment by s1monw** [comment]
> Shouldn't we also adapt the bwc tests so that it does not randomize the transport type when executing bwc tests against a future 6.x minor version (like testing 6.1.0 against 6.0.1)?

I don't understand what you mean, where is that randomization happening?

**Comment by tlrx** [comment]
@s1monw I was referring to [this line in ExternalNode](https://github.com/elastic/elasticsearch/blob/f5daa165f12a9ef84006ba16d89c8baf1efe4b94/test/framework/src/main/java/org/elasticsearch/test/ExternalNode.java#L55) - sorry I should have pointed to this line before.

**Comment by s1monw** [comment]
@tlrx this is dead code I think I will remove it soon we don't use this bwc test stuff these days

**Pull request merged**